execution_invariance_test/test.py

```diff
@@ -0,0 +1,154 @@
+
+
+def run_instrumented_project(project):
+    selected_hooks = get_hooks_from_analysis(["analysis.analysis.Analysis"])
+    project_dir = project.resolve()
+    project_src_dir = project_dir / "src"
+    for code_file in project_src_dir.rglob("*.py"):
+        instrument_file(str(project_dir / code_file), selected_hooks)
+
+    install_project(project)
+    json_report_file = project_dir / "instrumented_result.json"
+    result = pytest.main(["-v", "--cache-clear", "--json-report", "--json-report-file="+str(json_report_file), str(project_dir/"tests"), ])
+    with open(json_report_file, "r") as f:
+        result = f.read()
+    
+    result_json = json.loads(result)
+    test_result_map = {}
+    for item in result_json["tests"]:
+        test_result_map[item["nodeid"]] = item["outcome"]
+
+    # Clean up the project
+    for code_file in project_dir.rglob("*.py.orig"):
+            metadata_file = (
+                project_dir
+                / Path(*(code_file.parts[:-1]))
+                / (code_file.name[:-8] + "-dynapyt.json")
+            )
+            metadata_file.unlink()
+            correct_file = (
+                project_dir
+                / Path(*(code_file.parts[:-1]))
+                / (code_file.name[:-8] + ".py")
+            )
+            code_file.rename(correct_file)
+    json_report_file.unlink()
+
+    return test_result_map
+
+
+def remove_related_modules(project, project_module):
+    project_src_dir = project / "src"
+    for code_file in project_src_dir.rglob("*.py"): 
+        module_name = project_module + "." + code_file.stem
+        if module_name in sys.modules:
+            print(f"Deleting module {module_name}")
+            del sys.modules[module_name]
+
+    project_test_dir = project / "tests"
+    for code_file in project_test_dir.rglob("*.py"):
+        module_name = code_file.stem
+        if module_name in sys.modules:
+            print(f"Deleting module {module_name}")
+            del sys.modules[module_name]
+
+
+def run_tests(projects, project_module_map):
+    invariance_test_node_id = "tests/stack_test.py::test_stack"
+    failed = False
+    # Run the tests with and without instrumentation
+    for project in projects:
+        print(f"Installing project {project.name}")
+        install_project(project)
+        print(f"Running project {project.name}")
+        test_result = run_project(project)
+        remove_related_modules(project, project_module_map[project.name])
+        instrumented_test_result = run_instrumented_project(project)
+
+        for test, result in test_result.items():
+            if (test not in instrumented_test_result):
+                print(f"Test {test} not found in instrumented test results")
+                failed = True
+                break
+            if test == invariance_test_node_id:
+                if (result == instrumented_test_result[test]):
+                    print(f"Test {test} results match for invariance test")
```

Comment:
The message here should be more informative. Something like: "This test is expected to have different results from the uninstrumented code"